controllers/topology/internal/contract/metadata.go

```diff
@@ -42,7 +40,7 @@ func (m *Metadata) Get(obj *unstructured.Unstructured) (*clusterv1.ObjectMeta, e
 		return nil, errors.Wrap(err, "failed to retrieve control plane metadata.labels")
 	}
 	if !ok {
```

Comment:
@fabriziopandini  I agree that we shouldn't return an error here, but wouldn't it be more consistent overall and with e.g. mergeMap if we would return nil label/annotations if they are not set?
(the controller runtime client also sets empty maps to nil when it returns a resource)